core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -78,14 +78,17 @@ object RequestChannel extends Logging {
     val header: RequestHeader =
       if (requestObj == null) {
         buffer.rewind
-        RequestHeader.parse(buffer)
+        try RequestHeader.parse(buffer)
+        catch {
+          case ex: Throwable =>
+            throw new InvalidRequestException(s"Error parsing request header for apiKey: $requestId", ex)
```

Comment:
I love more information in the logs :)
The only point that worries me is that the API key is taken straight out of the header. If we failed to parse the header, the API Key may be straight out garbage. Perhaps clarify something along the lines of "Our best guess of the apiKey is..." - just to avoid being misleading?